### ISSUE TITLE
Numbered hints don't work when auto-follow is disabled

### ISSUE THREAD

**Issue by pkillnine** [qutebrowser/qutebrowser#2227]
Pressing enter doesn't follow the hint.

**Comment by The-Compiler** [comment]
Indeed, it only resets the filtered hints it seems.

Logs:

```
17:31:35 DEBUG    commands   command:run:509 command called: hint
17:31:35 DEBUG    commands   command:run:524 Calling qutebrowser.browser.hints.HintManager.start(<qutebrowser.browser.hints.HintManager object at 0x7fec38066168>, False, <Group.all: 1>, <Target.normal: 1>, win_id=0, mode=None, add_history=False)
17:31:35 DEBUG    hints      hints:_start_cb:580 hints: 00, 01, 02, 03, 04, 05, 06, 07, 08, 09, 10, 11, 12, 13, 14, 15, 16, 17
17:31:35 DEBUG    message    message:set_text:221 Follow hint...
17:31:35 DEBUG    statusbar  text:set_text:55 Setting normal text to 'Follow hint...'.
17:31:35 DEBUG    modes      modeman:enter:237 Entering mode KeyMode.hint (reason: HintManager.start)
17:31:35 DEBUG    modes      modeman:_eventFilter_keypress:191 handled: True, forward-unbound-keys: auto, passthrough: False, is_non_alnum: False --> filter: True (focused: <qutebrowser.browser.webkit.webview.WebView tab_id=0 url='https://github.com/'>)
17:31:35 DEBUG    modes      modeman:_eventFilter_keyrelease:211 filter: True
17:31:36 DEBUG    destroy    lineparser:_after_save:82 Saved to /tmp/qutebrowser-basedir-ly1_v4ya/data/cookies
17:31:37 DEBUG    modes      modeman:_eventFilter_keypress:166 got keypress in mode KeyMode.hint - delegating to <qutebrowser.keyinput.modeparsers.HintKeyParser supports_chains=True supports_count=False>
17:31:37 DEBUG    keyboard   basekeyparser:_debug_log:112 Got key: 0x30 / text: '0'
17:31:37 DEBUG    keyboard   basekeyparser:_debug_log:112 No match for '0' (added 0)
17:31:37 DEBUG    hints      hints:handle_partial_key:753 Handling new keystring: '0'
17:31:37 DEBUG    modes      modeman:_eventFilter_keypress:191 handled: True, forward-unbound-keys: auto, passthrough: False, is_non_alnum: False --> filter: True (focused: <qutebrowser.browser.webkit.webview.WebView tab_id=0 url='https://github.com/'>)
17:31:37 DEBUG    modes      modeman:_eventFilter_keyrelease:211 filter: True
17:31:37 DEBUG    modes      modeman:_eventFilter_keypress:166 got keypress in mode KeyMode.hint - delegating to <qutebrowser.keyinput.modeparsers.HintKeyParser supports_chains=True supports_count=False>
17:31:37 DEBUG    keyboard   basekeyparser:_debug_log:112 Got key: 0x33 / text: '3'
17:31:37 DEBUG    keyboard   basekeyparser:_debug_log:112 Definitive match for '03'.
17:31:37 DEBUG    keyboard   basekeyparser:_debug_log:112 discarding keystring '03'.
17:31:37 DEBUG    hints      hints:handle_partial_key:753 Handling new keystring: ''
17:31:37 DEBUG    hints      hints:handle_partial_key:753 Handling new keystring: '03'
17:31:37 DEBUG    modes      modeman:_eventFilter_keypress:191 handled: True, forward-unbound-keys: auto, passthrough: False, is_non_alnum: False --> filter: True (focused: <qutebrowser.browser.webkit.webview.WebView tab_id=0 url='https://github.com/'>)
17:31:37 DEBUG    modes      modeman:_eventFilter_keyrelease:211 filter: True
17:31:37 DEBUG    modes      modeman:_eventFilter_keypress:166 got keypress in mode KeyMode.hint - delegating to <qutebrowser.keyinput.modeparsers.HintKeyParser supports_chains=True supports_count=False>
'7:31:37 DEBUG    keyboard   basekeyparser:_debug_log:112 Got key: 0x1000004 / text: '
17:31:37 DEBUG    keyboard   basekeyparser:_debug_log:112 Ignoring, no text char
17:31:37 DEBUG    keyboard   modeparsers:_handle_special_key:168 Got special key 0x1000004 text 
17:31:37 DEBUG    modes      modeman:_eventFilter_keypress:191 handled: True, forward-unbound-keys: auto, passthrough: False, is_non_alnum: True --> filter: True (focused: <qutebrowser.browser.webkit.webview.WebView tab_id=0 url='https://github.com/'>)
17:31:37 DEBUG    modes      modeman:_eventFilter_keyrelease:211 filter: True
17:31:44 DEBUG    misc       app:on_focus_object_changed:768 Focus object changed: None
```

**Comment by lahwaacz** [comment]
Heh. [There is even a test for this](https://github.com/The-Compiler/qutebrowser/blob/master/tests/end2end/features/hints.feature#L412-L420), which still seems to be passing...

**Comment by lahwaacz** [comment]
And writing hints/numbers or text filter does not matter, neither in tests (where both seem to work) and practice (where it does not work).

**Comment by arza-zara** [comment]
According to my testing this is now fixed in 72e30cc12.

**Comment by jgkamat** [comment]
Yup, thanks for noticing! :]
